main.py

```diff
@@ -95,7 +95,10 @@ def main():
         save_path = os.path.join(timestamp,save_path)
     save_path = os.path.join(args.dataset,save_path)
     print('=> will save everything to {}'.format(save_path))
-    os.makedirs(save_path, exist_ok=True)
+    try:
+        os.makedirs(save_path)
+    except os.error: # if directory exists
+        pass
```

Comment:
It's certainly more elegant but alas, it does not exist in python 2. I'll write an if clause instead of try-except.